node-tests/e2e/test-support-test.js

```diff
@@ -62,34 +62,90 @@ describe('e2e: provides test support', function() {
     );
   });
 
-  afterEach(async function() {
-    await removeConfig(testProject);
-  });
+  describe('causes tests to fail on CSP violations', async function() {
+    afterEach(async function() {
+      await removeConfig(testProject);
+    });
 
-  it('causes tests to fail on CSP violations', async function() {
-    // runEmberCommand throws result object if command exists with non zero
-    // exit code
-    try {
-      await testProject.runEmberCommand('test');
+    it('causes tests to fail on CSP violations', async function() {
+      // runEmberCommand throws result object if command exists with non zero
+      // exit code
+      try {
+        await testProject.runEmberCommand('test');
 
-      // expect runEmberCommand to throw
-      expect(false).to.be.true;
-    } catch({ exitCode }) {
-      expect(exitCode).to.equal(1);
-    }
+        // expect runEmberCommand to throw
+        expect(false).to.be.true;
+      } catch({ exitCode }) {
+        expect(exitCode).to.equal(1);
+      }
+    });
+
+    it('does not cause tests failures if addon is disabled', async function() {
+      await setConfig(testProject, {
+        enabled: false,
+      });
+      let { exitCode } = await testProject.runEmberCommand('test');
+
+      expect(exitCode).to.equal(0);
+    });
+
+    it('does not cause tests failures if `failTests` config option is `false`', async function() {
+      await setConfig(testProject, {
+        failTests: false,
+      });
+
+      let { exitCode } = await testProject.runEmberCommand('test');
+
+      expect(exitCode).to.equal(0);
+    });
+
+    // One common scenario is when running the server in production mode locally, i.e.
+    // you are doing the production build on your local machine, connecting to your actual production server,
+    // for example via `ember serve -prod`. In these cases we don't want this addon to break.
+    it('does not break development server for builds not including tests', async function() {
+      // TODO: not supported yet
+      await testProject.startEmberServer({
+        environment: 'prodocution',
+        port: '49741',
+      });
+
+      let response = await request({
+        url: 'http://localhost:49741/',
+        headers: {
+          'Accept': 'text/html'
+        }
+      });
+      let responseForTests = await request({
+        url: 'http://localhost:49741/tests',
+        headers: {
+          'Accept': 'text/html'
+        }
+      });
+
+      expect(response.statusCode).to.equal(200);
+      expect(responseForTests.statusCode).to.equal(200);
+
+      await testProject.stopEmberServer();
+    });
```

Comment:
These tests are just moved from another position within the file. Beside that they are untouched if I recall correctly.